src/math/math.ts

```diff
@@ -1,106 +1,75 @@
 import marpitPlugin from '@marp-team/marpit/plugin'
-import katex from 'katex'
-import katexScss from './katex.scss'
-
-const convertedCSS = {}
-const katexMatcher = /url\(['"]?fonts\/(.*?)['"]?\)/g
+import * as katex from './katex'
+import * as mathjax from './mathjax'
 
 interface MathOptionsInterface {
+  lib?: 'katex' | 'mathjax'
   katexOption?: object
   katexFontPath?: string | false
 }
 
-export type MathOptions = boolean | MathOptionsInterface
-
-/**
- * marp-core math plugin
- *
- * It is implemented based on markdown-it-katex plugin. However, that is no
- * longer maintained by author. So we have ported math typesetting parser.
- *
- * @see https://github.com/waylonflinn/markdown-it-katex
- */
-export const markdown = marpitPlugin(
-  (md, updateState: (rendered: boolean) => void = () => {}) => {
-    const genOpts = (displayMode: boolean) => {
-      const math: MathOptions = md.marpit.options.math
-
-      return {
-        throwOnError: false,
-        ...(typeof math === 'object' && typeof math.katexOption === 'object'
-          ? math.katexOption
-          : {}),
-        displayMode,
-      }
-    }
+const contextSymbol = Symbol('marp-math-context')
 
-    md.core.ruler.before('block', 'marp_math_initialize', (state) => {
-      if (state.inlineMode) return
+export type MathOptions =
+  | boolean
+  | MathOptionsInterface['lib']
+  | MathOptionsInterface
 
-      updateState(false)
+export const markdown = marpitPlugin((md) => {
+  const opts: MathOptions | undefined = md.marpit.options.math
+  if (!opts) return
 
-      if (md.marpit.options.math) {
-        md.block.ruler.enable('marp_math_block')
-        md.inline.ruler.enable('marp_math_inline')
-      } else {
-        md.block.ruler.disable('marp_math_block')
```

Comment:
Marp specific options are frozen by Marpit so we will no longer consider changed options when rendering.